### ISSUE TITLE
Misplaced ``<div>`` in viewcode html output since Sphinx 1.7.0

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#4701]
Image shows it:

![capture d ecran 2018-03-02 a 15 13 27](https://user-images.githubusercontent.com/2589111/36902950-81851e38-1e2c-11e8-91fb-30f2bf1a4d92.png)

there is an extra line above the final one of the procedure, and the html source (in ``_build/html/_modules/foo.html``) looks like this

```html
<div class="viewcode-block" id="pgcd"><a class="viewcode-back" href="../index.html#foo.pgcd">[docs]</a><span class="k">def</span> <span class="nf">pgcd</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;Calcule le PGCD par l&#39;algorithme d&#39;Euclide.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">a</span><span class="p">,</span> <span class="n">b</span> <span class="o">=</span> <span class="nb">abs</span><span class="p">(</span><span class="n">a</span><span class="p">),</span> <span class="nb">abs</span><span class="p">(</span><span class="n">b</span><span class="p">)</span>
    <span class="k">while</span> <span class="n">b</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">a</span><span class="p">,</span> <span class="n">b</span> <span class="o">=</span> <span class="n">b</span><span class="p">,</span> <span class="n">a</span> <span class="o">%</span> <span class="n">b</span></div>
    <span class="k">return</span> <span class="n">a</span>
```

the closing ``</div>`` is mis-placed (not at end of last line but at end of previous one).

This does not happen with Sphinx 1.6.6 but happens with Sphinx 1.7.0 and 1.7.1

I am attaching 
[misplaceddiv.zip](https://github.com/sphinx-doc/sphinx/files/1775396/misplaceddiv.zip)
which one needs to compile with ``make html``.

### Environment info
- OS: Mac
- Running Sphinx v1.8.0+
- Sphinx version: 1.8.0+
- Python version: 3.6.4 (CPython)
- Docutils version: 0.14 
- Jinja2 version: 2.10

Also tested with 1.7.0 and 1.7.1. No problem with 1.6.6

**Comment by jfbu** [comment]
First bad commit appears to be 8683823536505aef3b96627ce9ea4784d5385eb0 (PR #3937)

**Comment by tk0miya** [comment]
Fixed by #4709
